doc/guides/modules.rst

```diff
@@ -10,14 +10,380 @@ system those are called *modules*. These modules must be integrated with the
+***********************************
+
+All source files in a module's code base shall include a license header,
+unless the module repository has **main license file** that covers source
+files that do not include license headers.
+
+Main license files shall be added in the module code base by Zephyr
+developers, only if they exist as part of the external project,
+and they contain a permissive OSI-compliant license. Main license files
+should preferably contain a full license text instead of including an
+SPDX license identifier. If multiple main license files are present it
+shall be made clear which license applies to each source file in a module
+code base.
+
+Individual license headers in module source files supersede the main license.
+
+Any new content to be added in a module repository will require to have
+license coverage.
+
+  .. note::
+
+     Zephyr recommends conveying module licensing via individual license
+     headers and main license files. This not a hard requirement; should
+     an external project have its own practice of conveying how licensing
+     applies in the module code base (for example, by having a single or
+     multiple main license files), this practice may be accepted by and
+     be referred to in the Zephyr module, as long as licensing requirement,
+     for example OSI compliance, are satisfied.
+
+License policies
+================
+
+When creating a module repository a developer shall:
+
+* import the main license files, if they exist in the external project, and
+* document (for example in the module README or .yml file) the default license
+  that covers the module code base.
+
+License checks
+--------------
+
+License checks (via CI tools) shall be enabled on every pull request that
+adds new content in module repositories.
+
+
+Documentation requirements
+**************************
+
+All Zephyr module repositories shall include an .rst file documenting:
+
+* the scope and the purpose of the module
+* how the module integrates with Zephyr
+* the owner of the module repository
+* synchronization information with the external project (commit, SHA, version etc.)
+* licensing information as described in :ref:`modules_licensing`.
+
+The file shall be required for the inclusion of the module and the contained
+information should be kept up to date.
+
+
+Testing requirements
+********************
+
+All Zephyr modules should provide some level of **integration** testing,
+ensuring that the integration with Zephyr works correctly.
+Integration tests:
+
+* may be in the form of a minimal set of samples and tests that reside
+  in the Zephyr main tree
+* should verify a basic usage of the module (configuration,
+  functional APIs, etc.) that is integrated with Zephyr.
+* shall be built and executed (for example in QEMU) as part of
+  sanitycheck runs in pull requests that introduce changes in module
+  repositories.
+
+  .. note::
+
+     New candidate modules shall provide some level of integration testing
```

Comment:
testing to be included in the manifest? This needs rephrasing I think

Reply:
Changed to:

```
     New modules, that are candidates for being included in the Zephyr default manifest,
     shall provide some level of integration testing.
```